discord/audit_logs.py

```diff
@@ -362,46 +360,19 @@ def _handle_app_command_permissions(
         before: AuditLogDiff,
         after: AuditLogDiff,
         entry: AuditLogEntry,
-        target_id: int,
         old_value: Optional[ApplicationCommandPermissions],
         new_value: Optional[ApplicationCommandPermissions],
     ):
-        guild = entry.guild
-
-        old_permission = new_permission = target = None
-
-        if target_id == (guild.id - 1):
-            # avoid circular import
-            from .app_commands import AllChannels
-
-            # all channels
-            target = AllChannels(guild)
-        else:
-            # get type and determine role, user or channel
-            _value = old_value or new_value
-            if _value is None:
-                return
-            permission_type = _value['type']
-            if permission_type == 1:
-                # role
-                target = guild.get_role(target_id)
-            elif permission_type == 2:
-                # user
-                target = entry._get_member(target_id)
-            elif permission_type == 3:
-                # channel
-                target = guild.get_channel(target_id)
-
-        if target is None:
-            target = Object(target_id)
+        # Avoid circular imports
+        from .app_commands.models import AppCommandPermissions
 
+        guild = entry.guild
+        state = entry._state
         if old_value is not None:
-            old_permission = old_value['permission']
-            before.app_command_permissions.append((target, old_permission))
+            before.app_command_permissions.append(AppCommandPermissions(data=old_value, guild=guild, state=state))
 
         if new_value is not None:
-            new_permission = new_value['permission']
-            after.app_command_permissions.append((target, new_permission))
+            after.app_command_permissions.append(AppCommandPermissions(data=new_value, guild=guild, state=state))
```

Comment:
This looks like it requires a documentation change for the audit log stuff.

Reply:
Yes, changes were made in [docs/api.rst](https://github.com/Rapptz/discord.py/pull/8086/files#diff-0ee295c29def16f570f86378eff8ba37315e464d415e731d5d8ea2e67ce16f68)